pkg/startup/startup_test.go

```diff
@@ -99,6 +100,19 @@ const (
 	randomULAPool = "<random ULA pool>"
 )
 
+// createClientset creates a kubernetes clientset when running under that platform.
+func createClientset() *kubernetes.Clientset {
```

Comment:
I don't think this is right - it relies on a real cluster being available (and also doesn't check for errors - it's just going to return a nil clientset IIUC). 

I think we should be using the fake client for UTs: https://godoc.org/k8s.io/client-go/kubernetes/fake

Reply:
(note my other comment above - I think we can remove the need to pass in a clientset to ensureDefaultConfig/configureIPPools simply by querying the configmap outside of those functions and passing in the configmap contents / boolean indicating if we're using kubeadm. that would remove dependence on the clientset here.)